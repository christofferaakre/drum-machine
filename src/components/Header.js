import React from 'react';
import theme from '../theme';

export const styles = {
  container: {
    position       : 'fixed',
    backgroundColor: theme.palette.primary.main,
    color          : 'white',
    padding        : 15,
    width          : '100%',
  },
  leftSide: {
    float: 'left',
  },
  rightSide: {
    float: 'right',
  },
  link: {
    display   : 'flex',
    alignItems: 'center',
  },
  iconSpan: {
    display   : 'flex',
    alignItems: 'center',
  },
  icon: {
    height: 26,
    margin: '0 5px',
  },
};

export const Header = () => (
  <header style={styles.container} id='header'>
    <div style={styles.leftSide}>
      <span>Drum machine by Christoffer Aakre</span>
    </div>
    <div style={styles.rightSide}>
      <a
        style={styles.link}
        className='link'
        href='https://github.com/christofferaakre/drum-machine'
      >
        Source on<span style={styles.iconSpan}>
          <img style={styles.icon} src='images/github-logo-light.png' alt='Github logo' /> Github
        </span>
      </a>
    </div>
  </header>
);

export default Header;
