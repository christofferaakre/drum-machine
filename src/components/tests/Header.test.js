import React from 'react';
import { shallow, } from 'enzyme';
import { Header, } from '../Header';

let props;

beforeEach(() => {
  props = {};
});

describe('Header', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
