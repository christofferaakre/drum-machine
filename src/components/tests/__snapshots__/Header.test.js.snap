// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  id="header"
  style={
    Object {
      "backgroundColor": "#009688",
      "color": "white",
      "padding": 15,
      "position": "fixed",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <span>
      Drum machine by Christoffer Aakre
    </span>
  </div>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <a
      className="link"
      href="https://github.com/christofferaakre/drum-machine"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      Source on
      <span
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <img
          alt="Github logo"
          src="images/github-logo-light.png"
          style={
            Object {
              "height": 26,
              "margin": "0 5px",
            }
          }
        />
         Github
      </span>
    </a>
  </div>
</header>
`;
